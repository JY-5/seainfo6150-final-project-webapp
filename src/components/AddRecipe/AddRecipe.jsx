import React from 'react';
import { useState } from 'react';
import styles from './AddRecipe.module.css';
import { Link, Redirect } from "react-router-dom";

const AddRecipe = (props) => {
    const [submittedForm, setSubmittedForm] = useState();
    function onSubmit(e) {
        e.preventDefault();
        const data = new FormData(e.target);
        setSubmittedForm(data);
    };

    return (
      <div className={styles.container}>
          {submittedForm ? (
            <div className={styles.addedRecipe}>
              <h1>Added new recipe</h1>
              <div>Title: {submittedForm.get("title")}</div>
              <div>Author: {props.username}</div>
              <div>Creation Date: {submittedForm.get("creationDate")}</div>
              <div>Category: {submittedForm.get("category")}</div>
              <div>Difficulty: {submittedForm.get("difficulty")}</div>
              <div>Preparation Time: {submittedForm.get("preparationTime")}</div>
              <div>Cost: {document.getElementById("cheap") && document.getElementById("cheap").checked ? "$" : 
                         (document.getElementById("affordable") && document.getElementById("affordable").checked ? "$$" :
                         (document.getElementById("expensive") && document.getElementById("expensive").checked ? "$$$" : "N/A"))
                }</div>
              <div>Vegetarian: {submittedForm.get("vegetarian") ? "Yes" : "No"}</div>
              <div>Ingredients: {submittedForm.get("ingredients")}</div>
              <div>Utensils: {submittedForm.get("utensils")}</div>
              <div>Instructions: {submittedForm.get("instructions")}</div>
              <Link to="/" className={styles.link}>Okay</Link>
            </div>
          ) : (
            <div className={styles.addPanel}>
            <div className={styles.toAdd}>
              <h1>Add a new recipe</h1>
              <form onSubmit={onSubmit} className={styles.form}>
                <div className={styles.title}>
                  <label htmlFor="title">Title*</label>
                  <input type="text" name="title" id="title" required/>
                </div>
                <div className={styles.creationDate}>
                  <label htmlFor="title">Creation Date</label>
                  <input type="date" name="creationDate" id="creationDate" size="10"/>
                </div>
                <div className={styles.category}>
                  <label htmlFor="category">Category</label>
                  <select name="category" id="category">
                    <option value="American">American</option>
                    <option value="Chinese">Chinese</option>
                    <option value="Japanese">Japanese</option>
                    <option value="Korean">Korean</option>
                    <option value="Mexican">Mexican</option>
                    <option value="Thai">Thai</option>
                    <option value="Vietnamese">Vietnamese</option>
                    <option value="Other">Other</option>
                  </select>
                </div>
                <div className={styles.difficulty}>
                  <label htmlFor="difficulty">Difficulty</label>
                  <select name="difficulty" id="difficulty">
                    <option value="Beginner">Beginner</option>
                    <option value="Intermediate">Intermediate</option>
                    <option value="Expert">Expert</option>
                  </select>
                </div>
                <div className={styles.preparationTime}>
                  <label htmlFor="preparationTime">Preparation Time (Minutes)</label>
                  <input type="number" name="preparationTime" id="preparationTime"/>
                </div>
                <div className={styles.cost}>
                  <label htmlFor="cost">Cost</label>
                  <input type="radio" name="cost" id="cheap" value="1"/>
                  <label htmlFor="cheap">$</label>
                  <input type="radio" name="cost" id="affordable" value="2"/>
                  <label htmlFor="affordable">$$</label>
                  <input type="radio" name="cost" id="expensive" value="3"/>
                  <label htmlFor="expensive">$$$</label>
                </div>
                <div className={styles.vegetarian}>
                  <input type="checkbox" name="vegetarian" id="vegetarian" value="1"/>
                  <label htmlFor="vegetarian">Vegetarian</label>
                </div>
                <div className={styles.ingredients}>
                  <label htmlFor="ingredients">Ingredients*</label>
                  <input type="text" name="ingredients" id="ingredients" required/>
                </div>
                <div className={styles.utensils}>
                  <label htmlFor="utensils">Utensils</label>
                  <input type="text" name="utensils" id="utensils"/>
                </div>
                <div className={styles.instructions}>
                  <label htmlFor="instructions">Instructions*</label>
                  <textarea name="instructions" id="instructions" required/>
                </div>
                <div className={styles.addButton}>
                  <input type="submit" value="Add" />
                </div>
              </form>
            </div>
            </div>
          )
          }
      </div>
    )
}

export default AddRecipe
