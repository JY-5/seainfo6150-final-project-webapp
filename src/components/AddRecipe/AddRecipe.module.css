.container {
    /* width: 80%; */
    font-family: 'Rubik', sans-serif;
    margin-left: 2rem;
    margin-right: 2rem;
    /* width: 200rem; */
}

.container div {
    margin-bottom: 0.7rem;
}

.addPanel {
    display: flex;
    justify-content: center;
}

.toAdd {
    width: 60rem;
}

.container h1 {
    color: #539222;
}

.container label {
    margin-right: 0.5rem;
}

.addedRecipe {
    display: flex;
    flex-direction: column;
    align-items: stretch;
    margin: 0 2rem 1rem;
}

.link {
    text-decoration: none;
    background-color: #6eb339;
    padding: 0.5rem;
    color: white;
    font-weight: 600;
    font-size: 1.5rem;
    border-radius: 5px;
    width: 8rem;
    text-align: center;
    margin: 1rem auto;
}

.link:hover {
    background-color: #539222;
}

.instructions {
    height: auto;

}
.form {
    /* display: flex;
    flex-direction: column; */
}

.title {
    display: flex;
}

.title input {
    flex: 1;
}

.ingedients {
    display: flex;
}

.ingrediens input {
    flex: 1;
}

.utensils {
    display: flex;
}

.utensils input {
    flex: 1;
}

.instructions {
    display: flex;
    flex-flow: column;
}

.instructions textarea {
    height: 30rem;
}

.category {

}

.addButton {
    /* flex: 1; */
}

.addButton input {
    width: 100%;
    padding: 0.8rem;
    background-color: #6eb339;
    color: white;
    font-weight: 600;
    font-size: 1rem;
    border: none;
    border-radius: 5px;
}

.addButton input:hover {
    background-color: #539222;
}