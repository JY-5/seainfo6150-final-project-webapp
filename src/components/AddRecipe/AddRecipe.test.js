import React from "react";
import AddRecipe from './AddRecipe';
import { render } from "@testing-library/react";
import {
  BrowserRouter as Router,
} from "react-router-dom";

describe("Add recipe tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Router><AddRecipe username="JY"/></Router>);
      expect(container).toMatchSnapshot();
    });
});