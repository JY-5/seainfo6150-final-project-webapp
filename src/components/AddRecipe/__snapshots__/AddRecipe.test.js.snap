// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add recipe tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <div
        class="form"
      >
        <h1>
          Add a new recipe
        </h1>
        <form
          class="form"
        >
          <div
            class="title"
          >
            <label
              for="title"
            >
              Title*
            </label>
            <input
              id="title"
              name="title"
              required=""
              size="50"
              type="text"
            />
          </div>
          <div
            class="creationDate"
          >
            <label
              for="title"
            >
              Creation Date
            </label>
            <input
              id="creationDate"
              name="creationDate"
              size="10"
              type="date"
            />
          </div>
          <div
            class="category"
          >
            <label
              for="category"
            >
              Category
            </label>
            <select
              id="category"
              name="category"
            >
              <option
                value="American"
              >
                American
              </option>
              <option
                value="Chinese"
              >
                Chinese
              </option>
              <option
                value="Japanese"
              >
                Japanese
              </option>
              <option
                value="Korean"
              >
                Korean
              </option>
              <option
                value="Mexican"
              >
                Mexican
              </option>
              <option
                value="Thai"
              >
                Thai
              </option>
              <option
                value="Vietnamese"
              >
                Vietnamese
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
          </div>
          <div
            class="difficulty"
          >
            <label
              for="difficulty"
            >
              Difficulty
            </label>
            <select
              id="difficulty"
              name="difficulty"
            >
              <option
                value="Beginner"
              >
                Beginner
              </option>
              <option
                value="Intermediate"
              >
                Intermediate
              </option>
              <option
                value="Expert"
              >
                Expert
              </option>
            </select>
          </div>
          <div
            class="preparationTime"
          >
            <label
              for="preparationTime"
            >
              Preparation Time (Minutes)
            </label>
            <input
              id="preparationTime"
              name="preparationTime"
              type="number"
            />
          </div>
          <div
            class="cost"
          >
            <label
              for="cost"
            >
              Cost
            </label>
            <input
              id="cheap"
              name="cost"
              type="radio"
              value="1"
            />
            <label
              for="cheap"
            >
              $
            </label>
            <input
              id="affordable"
              name="cost"
              type="radio"
              value="2"
            />
            <label
              for="affordable"
            >
              $$
            </label>
            <input
              id="expensive"
              name="cost"
              type="radio"
              value="3"
            />
            <label
              for="expensive"
            >
              $$$
            </label>
          </div>
          <div
            class="vegetarian"
          >
            <input
              id="vegetarian"
              name="vegetarian"
              type="checkbox"
              value="1"
            />
            <label
              for="vegetarian"
            >
              Vegetarian
            </label>
          </div>
          <div
            class="ingredients"
          >
            <label
              for="ingredients"
            >
              Ingredients*
            </label>
            <input
              id="ingredients"
              name="ingredients"
              required=""
              size="100"
              type="text"
            />
          </div>
          <div
            class="utensils"
          >
            <label
              for="utensils"
            >
              Utensils
            </label>
            <input
              id="utensils"
              name="utensils"
              size="100"
              type="text"
            />
          </div>
          <div
            class="instructions"
          >
            <label
              for="instructions"
            >
              Instructions*
            </label>
            <textarea
              cols="100"
              id="instructions"
              name="instructions"
              required=""
              rows="10"
            />
          </div>
          <div
            class="addButton"
          >
            <input
              type="submit"
              value="Add"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
